src/absil/il.fs

```diff
@@ -1400,19 +1409,50 @@ type ILMethodDef =
     member x.MaxStack     = x.MethodBody.MaxStack  
     member x.IsZeroInit   = x.MethodBody.IsZeroInit
 
-    member x.IsClassInitializer   = match x.mdKind with | MethodKind.Cctor      -> true | _ -> false
-    member x.IsConstructor        = match x.mdKind with | MethodKind.Ctor       -> true | _ -> false
-    member x.IsStatic             = match x.mdKind with | MethodKind.Static     -> true | _ -> false
-    member x.IsNonVirtualInstance = match x.mdKind with | MethodKind.NonVirtual -> true | _ -> false
-    member x.IsVirtual            = match x.mdKind with | MethodKind.Virtual _  -> true | _ -> false
-
-    member x.IsFinal                = match x.mdKind with | MethodKind.Virtual v -> v.IsFinal    | _ -> invalidOp "not virtual"
-    member x.IsNewSlot              = match x.mdKind with | MethodKind.Virtual v -> v.IsNewSlot  | _ -> invalidOp "not virtual"
-    member x.IsCheckAccessOnOverride= match x.mdKind with | MethodKind.Virtual v -> v.IsCheckAccessOnOverride   | _ -> invalidOp "not virtual"
-    member x.IsAbstract             = match x.mdKind with | MethodKind.Virtual v -> v.IsAbstract | _ -> invalidOp "not virtual"
-
     member md.CallingSignature =  mkILCallSig (md.CallingConv,md.ParameterTypes,md.Return.Type)
 
+    member x.IsClassInitializer   = x.Name = ".cctor"
+    member x.IsConstructor        = x.Name = ".ctor"
+
+    member x.Access                 = memberAccessOfFlags (int x.Attributes)
+    member x.IsStatic               = x.Attributes &&& MethodAttributes.Static <> enum 0
+    member x.IsNonVirtualInstance   = not x.IsStatic && not x.IsVirtual
```

Comment:
Looks like this is not used anywhere in the codebase, better to delete it I think

Reply:
@dsyme it's used in https://github.com/Microsoft/visualfsharp/pull/4272/files/9e2e6f0887e022b2ff73e79bf7aec02b00c77d10#diff-75b9faf38cb7c46bae44ad1d6d43496bR800